### PR TITLE
Add tests coverage for the opt components functionality

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

Changes proposed in this pull request:

- Add tests coverage for the opt components functionality

**Related issue(s)**

- https://github.com/kyma-incubator/compass/issues/723

### CODE REVIEW

File: components/kyma-environment-broker/internal/runtime/provider_test.go [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+				managedRuntimeComponentsYAMLPath: path.Join("testdata", "managed-runtime-components.yaml"),
+				httpErrMessage:                   "Not Found",
+			},
+			expErrMessage: "while getting open source Kyma components list: got unexpected status code, want 200, got 404, url: https://storage.googleapis.com/kyma-development-artifacts/master-123123/kyma-installer-cluster.yaml, body: Not Found",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// given
+			fakeHTTPClient := newTestClient(t, tc.given.httpErrMessage, http.StatusNotFound)
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/runtime/provider_test.go]: why didn't you use "net/http/httptest" package?

File: components/kyma-environment-broker/internal/runtime/provider.go [hunk 2]
```diff
@@ -24,11 +24,17 @@ const (
...
+	httpClient                       HTTPDoer
+}
+
+type HTTPDoer interface {
+	Do(req *http.Request) (*http.Response, error)
 }
 
 // NewComponentsListProvider returns new instance of the ComponentsListProvider
-func NewComponentsListProvider(kymaVersion string, managedRuntimeComponentsYAMLPath string) *ComponentsListProvider {
+func NewComponentsListProvider(client HTTPDoer, kymaVersion string, managedRuntimeComponentsYAMLPath string) *ComponentsListProvider {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/runtime/provider.go]: if there is really no way to do it without injecting http doer please create separate contructor for tests

File: components/kyma-environment-broker/internal/runtime/provider.go [hunk 3]
```diff
@@ -61,11 +67,15 @@ func (r ComponentsListProvider) AllComponents() ([]v1alpha1.KymaComponent, error
 func (r ComponentsListProvider) getOpenSourceKymaComponents() (comp []v1alpha1.KymaComponent, err error) {
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/internal/runtime/provider.go]: what about pointer receiver?

File: components/kyma-environment-broker/internal/runtime/provider_test.go (hunk 1 above)
Comment (mszostok) [reply components/kyma-environment-broker/internal/runtime/provider_test.go]: the httptest gives you only http server. To use that you will need to inject the URL to provider and in current impl it is a provider responsibility to do that, so I can also assert that 
```go
assert.Equal(t, tc.expectedRequestURL, fakeHTTPClient.RequestURL)
```

So better is to use client and assert real request data

File: components/kyma-environment-broker/internal/runtime/provider.go (hunk 2 above)
Comment (mszostok) [reply components/kyma-environment-broker/internal/runtime/provider.go]: I've added dedicated setter available only for tests

**Pull request merged**